### PR TITLE
chore: bump 'expect' dev dep from v26 to v27

### PR DESCRIPTION
**Pull request by mrienstra**
Motivation: https://github.com/facebook/jest/issues/11640

This change causes `npm test` to fail, because of a problem when running `npm run check-deps`, which runs `utils/check_deps.js`.

More specifically, the function `allowExternalImport` / `alllowExternalImport` (in `utils/check_deps.js`) -- which was very recently added in #8501 -- is failing, when looking at the usage of the `expect` dev dep.

I played around a little with `node_modules/expect/package.json` locally to see what kinds of changes would resolve this. 

Looks like removing the `"exports"` section does the trick (it was added in https://github.com/facebook/jest/pull/9921), but that's not much of a solution.

Adding:
```
    "./build/types": "./build/types.d.ts",
    "./build/jasmineUtils": "./build/jasmineUtils.js",
    "./build/matchers": "./build/matchers.js"
    "./build/print": "./build/print.js",
```
... Also does the job, but is presumably unnecessarily specific.

Adding:
```
    "./build/types": "./build/types.d.ts",
    "./build/*": "./build/*.js"
```
... Is probably the right solution.
(Bonus: this would also allow removing this existing line: `"./build/utils": "./build/utils.js",`)

Opened a PR to make this change: https://github.com/facebook/jest/pull/11816

Just in case there is a better path forward, opening this draft PR to allow conversation to take place both here and in the `jest` PR. @mxschmitt, @pavelfeldman, thoughts?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -61,7 +61,7 @@
     "colors": "^1.4.0",
     "commander": "^6.1.0",
     "debug": "^4.1.1",
-    "expect": "^26.4.2",
+    "expect": "^27.1.0",
     "extract-zip": "^2.0.1",
     "https-proxy-agent": "^5.0.0",
     "jest-matcher-utils": "^26.4.2",
```
Comment (mxschmitt) [inline package.json:67]: please also bump the jest-matcher-utils

**Comment by mrienstra** [comment]
FYI I ran tests locally (with local fix to `node_modules/expect/package.json` to make `utils/check_deps.js` happy), and everything passed except:

**A)** `[webkit] › proxy.spec.ts:142:1 › should use socks proxy` & `[webkit] › proxy.spec.ts:153:1 › should use socks proxy in second page` failed often -- but not always -- with:
```
    page.goto: The network connection was lost.
    =========================== logs ===========================
    navigating to "http://non-existent.com/", waiting until "load"
    ============================================================

      146 |   });
      147 |   const page = await browser.newPage();
    > 148 |   await page.goto('http://non-existent.com');
```

**B)** `selectors-register.spec.ts:20:1 › should work` failed once (`Timeout of 30000ms exceeded`, for all 3 browsers) when running all tests, but passed during 3 re-runs of `npm test ./tests/selectors-register.spec.ts`

**C)** `[firefox] › inspector/pause.spec.ts:236:3 › pause should populate log with error in waitForEvent` failed once when running all tests, but passed during 3 re-runs of `npm test ./tests/inspector/pause.spec.ts`

Error was:
```
    Error: expect(received).toEqual(expected) // deep equality

    - Expected  -  1
    + Received  + 11

      Array [
        "page.pause- XXms",
        "page.waitForEvent(console)",
        "waiting for event \"console\"",
        "error: Timeout while waiting for event \"console\"",
    -   "page.click(button)- XXms",
    +   "page.click(button)",
    +   "waiting for selector \"button\"",
    +   "selector resolved to visible <button>Submit</button>",
    +   "attempting click action",
    +   "waiting for element to be visible, enabled and stable",
    +   "element is visible, enabled and stable",
    +   "scrolling into view if needed",
    +   "done scrolling",
    +   "checking that element receives pointer events at (36.1,19)",
    +   "element does receive pointer events",
    +   "performing click action",
      ]
```

None of these test failures appear to be related to the changes in this PR.

File: package.json (hunk 1 above)
Comment (mrienstra) [reply package.json]: Done!

**Comment by mrienstra** [comment]
Alternate fix, shown in https://github.com/microsoft/playwright/pull/8718/commits/7e26ebb: add `'expect/build/types', 'expect/build/jasmineUtils', 'expect/build/print', 'expect/build/matchers'` to `EXTERNAL_IMPORT_ALLOWLIST`.

This doesn't seem like an ideal solution, but it decouples this PR from https://github.com/facebook/jest/pull/11816, which is nice.

**Comment by mrienstra** [comment]
Extending `EXTERNAL_IMPORT_ALLOWLIST` doesn't seem to be a solution, just pushes the problem further down:
```
node:internal/modules/cjs/loader:484
      throw e;
      ^

Error [ERR_PACKAGE_PATH_NOT_EXPORTED] [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './build/jasmineUtils' is not defined by "exports" in /Users/mikey/Sites/playwright/node_modules/expect/package.json
    at new NodeError (node:internal/errors:363:5)
    at throwExportsNotFound (node:internal/modules/esm/resolve:321:9)
    at packageExportsResolve (node:internal/modules/esm/resolve:546:3)
    at resolveExports (node:internal/modules/cjs/loader:478:36)
    at Function.Module._findPath (node:internal/modules/cjs/loader:518:31)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:927:27)
    at Function.Module._load (node:internal/modules/cjs/loader:774:27)
    at Module.require (node:internal/modules/cjs/loader:1013:19)
    at require (node:internal/modules/cjs/helpers:93:18)
    at Object.<anonymous> (/Users/mikey/Sites/playwright/lib/test/matchers/toEqual.js:8:21)
    at Module._compile (node:internal/modules/cjs/loader:1109:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1138:10)
    at Module.load (node:internal/modules/cjs/loader:989:32)
    at Function.Module._load (node:internal/modules/cjs/loader:829:14)
    at Module.require (node:internal/modules/cjs/loader:1013:19) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}
```
... So I reverted those changes.

**Comment by mrienstra** [comment]
Looking for someone who knows the `playwright` codebase well to take a quick look at:
https://github.com/facebook/jest/pull/11816
... And chime in if they have anything important to add to that conversation. Thanks!

@pavelfeldman, @dgozman, @aslushnikov, @yury-s, @JoelEinbinder, @mxschmitt
(Sorry if this is redundant, but not sure if @-mentions on repos not "watched" will be seen)

**Comment by mxschmitt** [comment]
@mrienstra since we dropped `jest-matcher-utils` in #8782 it should work to bump expect to 27 after you rebase. Thanks for your effort!

**Review by mxschmitt — APPROVED** [review]

**Review by JoelEinbinder — APPROVED** [review]

**Pull request merged**